drivers/FileHandle.h

```diff
@@ -25,62 +25,85 @@ namespace mbed {
 /** \addtogroup drivers */
 /** @{*/
 
-/** An OO equivalent of the internal FILEHANDLE variable
+
+/** Class FileHandle
+ *
+ *  An OO equivalent of the internal FILEHANDLE variable
  *  and associated _sys_* functions.
  *
- * FileHandle is an abstract class, needing at least sys_write and
+ *  FileHandle is an abstract class, needing at least sys_write and
```

Comment:
Yeah, this comment was pulled in from the previous version and is kinda inaccurate, will update.

A file with only len? What would that be used for? I guess it's possible.

It could make sense to have other interfaces for devices with different limitations (ie ReadStream, Seekable), but after expirementing with the idea, it doesn't seem to fit in with c++'s oop inheritance well and brought little benefit outside of documentation.